windows/src/universal_ble_plugin.cpp

```diff
@@ -538,6 +505,209 @@ namespace universal_ble
+        {
+            deviceAddressKey,
+            isConnectedKey,
+            isPairedKey,
+            isPresentKey,
+            isConnectableKey,
+            signalStrengthKey,
+        },
+        DeviceInformationKind::AssociationEndpoint);
+
+    /// Device Added from DeviceWatcher
+    deviceWatcherAddedToken = deviceWatcher.Added([this](DeviceWatcher sender, DeviceInformation deviceInfo)
+                                                  {
+                                                    std::string deviceId = winrt::to_string(deviceInfo.Id());
+                                                    deviceWatcherDevices.insert_or_assign(deviceId, deviceInfo);
+                                                    onDeviceInfoRecieved(deviceInfo);
+                                                    // On Device Added
+                                                  });
+
+    // Update only if device is already discovered in deviceWatcher.Added
+    deviceWatcherUpdatedToken = deviceWatcher.Updated([this](DeviceWatcher sender, DeviceInformationUpdate deviceInfoUpdate)
+                                                      {
+                                                        std::string deviceId = winrt::to_string(deviceInfoUpdate.Id());
+                                                        auto it = deviceWatcherDevices.find(deviceId);
+                                                        if (it != deviceWatcherDevices.end())
+                                                        {
+                                                          it->second.Update(deviceInfoUpdate);
+                                                          onDeviceInfoRecieved(it->second);
+                                                        }
+                                                        // On Device Updated
+                                                      });
+
+    deviceWatcherRemovedToken = deviceWatcher.Removed([this](DeviceWatcher sender, DeviceInformationUpdate args)
+                                                      {
+                                                        std::string deviceId = winrt::to_string(args.Id());
+                                                        deviceWatcherDevices.erase(deviceId);
+                                                        // On Device Removed
+                                                      });
+  }
+
+  void UniversalBlePlugin::disposeDeviceWatcher()
+  {
+    if (deviceWatcher != nullptr)
+    {
+      if (deviceWatcher.Status() == DeviceWatcherStatus::Started)
+        deviceWatcher.Stop();
+      deviceWatcher.Added(deviceWatcherAddedToken);
+      deviceWatcher.Updated(deviceWatcherUpdatedToken);
+      deviceWatcher.Removed(deviceWatcherRemovedToken);
+      deviceWatcher = nullptr;
+      // Dispose tokens
+      deviceWatcherDevices.clear();
+    }
+  }
+
+  void UniversalBlePlugin::onDeviceInfoRecieved(DeviceInformation deviceInfo)
```

Comment:
```suggestion
  void UniversalBlePlugin::onDeviceInfoReceived(DeviceInformation deviceInfo)
```
We should spell-check the files